### PR TITLE
[Merged by Bors] - Update to consensus-specs v1.1.9

### LINKED ISSUE #3014: Update to spec v1.1.9
## Description

We should update to spec [v1.1.9](https://github.com/ethereum/consensus-specs/releases/tag/v1.1.9) to suit the [Kiln spec](https://hackmd.io/@n0ble/kiln-spec).

A good place to start is bumping the flag on the tests and seeing what breaks:

https://github.com/sigp/lighthouse/blob/0177b9286edfdeb2782f74fc7fb1392b6459bfaa/testing/ef_tests/Makefile#L1

### LINKED ISSUE #3015: Implement authentication for execution API
## Description

The authentication scheme is documented here: https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.7/src/engine/authentication.md

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

Closes #3014

## Proposed Changes

- Rename `receipt_root` to `receipts_root`
- Rename `execute_payload` to `notify_new_payload`
   - This is slightly weird since we modify everything except the actual HTTP call to the engine API. That change is expected to be implemented in #2985 (cc @ethDreamer)
- Enable "random" tests for Bellatrix.

## Notes

This will break *partially* compatibility with Kintusgi testnets in order to gain compatibility with [Kiln](https://hackmd.io/@n0ble/kiln-spec) testnets. I think it will only break the BN APIs due to the `receipts_root` change, however it might have some other effects too.

### CODE REVIEW

**Comment by paulhauner** [comment]
I have updated the [`kintsugi`](https://github.com/sigp/lighthouse/tree/kintsugi) branch to the current `unstable` commit: https://github.com/sigp/lighthouse/commit/b5921e4248fae19e874da595db55a124f22a538a

That should provide a frozen version that works with Kintsugi.

**Comment by realbigsean** [comment]
Do we need to add `CONFIG_NAME` ? https://github.com/ethereum/consensus-specs/pull/2784/files

File: beacon_node/beacon_chain/src/execution_payload.rs
Comment (realbigsean) [inline beacon_node/beacon_chain/src/execution_payload.rs]:
```suggestion
/// https://github.com/ethereum/consensus-specs/blob/v1.1.9/specs/bellatrix/beacon-chain.md#notify_new_payload
```

**Comment by paulhauner** [comment]
> Do we need to add `CONFIG_NAME` ? https://github.com/ethereum/consensus-specs/pull/2784/files

Good point! I might leave that one for now and @michaelsproul is going to follow up.

**Comment by paulhauner** [comment]
~~Blocked on `v2.1.3` release.~~

**Comment by michaelsproul** [comment]
Opened a PR to this branch here: https://github.com/paulhauner/lighthouse/pull/2

We can either merge it down to this branch or re-open it as a separate PR if this PR merges first

**Comment by paulhauner** [comment]
Thanks @michaelsproul :pray: All good to go now, pending CI.

**Review by michaelsproul — APPROVED** [review]
Looks good

**Pull request closed without merging**
